### PR TITLE
dotCMS/core#21991 Add table block 

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Add table block to block editor, https://github.com/dotcms/core/issues/21991

### CODE REVIEW

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss [hunk 1]
```diff
@@ -139,6 +139,96 @@
...
+            table {
+                border-collapse: collapse;
+                margin: 0;
+                overflow: hidden;
+                table-layout: fixed;
+                width: 100%;
+
+                td,
+                th {
+                    border: 2px solid #ced4da;
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss:152]: vars

File: core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.html [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+                    #group
+                    *ngSwitchDefault
+                    [formControlName]="control.key"
+                    [id]="control.key"
+                    [type]="control.type"
+                    [min]="control.min"
+                    [style]="{ width: '100%', fontSize: '14px', height: '40px' }"
+                    pInputText
+                    type="control.type"
+                />
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.html:24]: Shouldn't we wrap this in a `<div class="field">` like we do in the dotcms-ui forms?

File: core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.scss [hunk 3]
```diff
@@ -24,6 +32,11 @@ form {
     label {
         font-size: 0.75 * $dot-editor-size;
     }
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.scss:34]: We should use the global styling forms fields

File: core-web/libs/block-editor/src/lib/extensions/bubble-menu/plugins/dot-bubble-menu.plugin.ts [hunk 4]
```diff
@@ -330,6 +333,47 @@ export class DotBubbleMenuPluginView extends BubbleMenuView {
...
+                                  controlType: 'text',
+                                  type: 'text'
+                              }
+                          ])
+                          .pipe(
+                              take(1),
+                              filter((data) => data != null)
+                          )
+                          .subscribe((data) => {
+                              requestAnimationFrame(() => {
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/extensions/bubble-menu/plugins/dot-bubble-menu.plugin.ts:372]: Why we need this?

File: core-web/libs/block-editor/src/lib/extensions/dot-table/dot-table-cell.plugin.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            handleDOMEvents: {
+                contextmenu: (view, event) => {
+                    const grandpaSelectedNode = view.state.selection.$from.node(
+                        view.state.selection.$from.depth - 1
+                    );
+
+                    if (
+                        grandpaSelectedNode.type.name === 'tableCell' ||
+                        grandpaSelectedNode.type.name === 'tableHeader' ||
+                        grandpaSelectedNode.type.name === 'tableRow'
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/extensions/dot-table/dot-table-cell.plugin.ts:134]: Move this to a reusable function

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.extension.ts [hunk 6]
```diff
@@ -49,15 +58,17 @@ export const BubbleFormExtension = (viewContainerRef: ViewContainerRef) => {
...
+                        chain()
                             .command(({ tr }) => {
-                                tr.setMeta(BUBBLE_FORM_PLUGIN_KEY, { open: true });
+                                tr.setMeta(BUBBLE_FORM_PLUGIN_KEY, { form, options, open: true });
 
                                 return true;
                             })
                             .run();
+
+                        return formValue$;
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.extension.ts:71]: I'm still a little worried about this. Should we go through the block editor test cases to see that this doesn't break anything?

File: core-web/libs/block-editor/src/lib/shared/components/suggestion-list/components/suggestions-list-item/suggestions-list-item.component.ts [hunk 7]
```diff
@@ -28,7 +29,9 @@ export class SuggestionsListItemComponent implements FocusableOption, OnInit {
     @HostListener('mousedown', ['$event'])
     onMouseDown(e: MouseEvent) {
         e.preventDefault();
-        this.command();
+        if (!this.disabled) {
+            this.command();
+        }
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/shared/components/suggestion-list/components/suggestions-list-item/suggestions-list-item.component.ts:34]: Why do we need this?

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/libs/block-editor/src/lib/extensions/dot-table/dot-table-cell.extension.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        renderHTML({ HTMLAttributes }) {
+            return [
+                'td',
+                mergeAttributes(this.options.HTMLAttributes, HTMLAttributes),
+                [
+                    'button',
+                    {
+                        class: 'dot-cell-arrow'
+                    }
+                ],
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/dot-table/dot-table-cell.extension.ts:17]: correct, the button is not present.

File: core-web/libs/block-editor/src/lib/shared/components/suggestion-list/components/suggestions-list-item/suggestions-list-item.component.ts (hunk 7 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/shared/components/suggestion-list/components/suggestions-list-item/suggestions-list-item.component.ts:34]: to ignore the command execution when the option is disabled.

File: core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.html (hunk 2 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.html:24]: we use `class="form-row"` in this case. not needed.

File: core-web/libs/block-editor/src/lib/extensions/dot-table/dot-table-cell.plugin.ts [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+    init(): void {}
+
+    // eslint-disable-next-line
+    update(): void {}
+
+    destroy() {
+        this.tippy.destroy();
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/dot-table/dot-table-cell.plugin.ts:26]: to used the destroy the other methods are mandatory.

File: core-web/libs/block-editor/src/lib/extensions/bubble-menu/plugins/dot-bubble-menu.plugin.ts (hunk 4 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/bubble-menu/plugins/dot-bubble-menu.plugin.ts]: wait the tippy form to render.

File: core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.html (hunk 2 above)
Comment (fmontes) [reply core-web/libs/block-editor/src/lib/extensions/bubble-form/bubble-form.component.html:24]: We do need it... the thing is that we should use the same class of the forms we have in `dotcms-ui` to keep things consistent and for maintainability.

**Pull request merged**